numpy/f2py/cfuncs.py

```diff
@@ -479,20 +479,43 @@
 cppmacros['STRINGFREE'] = """\
 #define STRINGFREE(str) do {if (!(str == NULL)) free(str);} while (0)
 """
+needs['STRINGPADN'] = ['string.h']
+cppmacros['STRINGPADN'] = """\
+/*
+STRINGPADN replaces nulls with padding from the right.
+
+`to` must have size of at least N bytes.
+
+If the `to[N-1]` is null (`\\0`), then replace it and all the
+preceeding nulls with the given padding.
+*/
+#define STRINGPADN(to, N, PADDING)                              \\
+    do {                                                        \\
+        int _m = (N);                                           \\
+        char *_to = (to);                                       \\
+        FAILNULL(_to);                                          \\
+        for (_m -= 1; _m >= 0 && _to[_m] == '\\0'; _m--) {      \\
+             _to[_m] = PADDING;                                 \\
+        }                                                       \\
+    } while (0)
+"""
```

Comment:
This looks good, but also doesn't seem to be used anywhere!

Reply:
Removed STRINGPADN as unused.